### DISCUSSION TITLE
Is there a public Hyper gateway?

### DISCUSSION THREAD

**Discussion by akavel** [beakerbrowser/beaker#1685] [General]
Is there some kind of a "public service" HTTP(S) mirror of the "hyper-web" (?), where non-Beaker people could easily browse any contents I created with Beaker, similar as with the IPFS gateways?

**Comment by pfrazee** [comment]
Not to my knowledge, but somebody in the community might make one eventually.

**Comment by brechtcs** [comment]
@reaktivo had something like that deployed at https://hyper.reaktivo.com last week, although it seems to be down at the moment.

**Comment by reaktivo** [comment]
@akavel Yes, there's hyper.reaktivo.com but you have to consider it's not supposed to be depended on, since I haven't managed to have the hyperdrive-daemon to work for a few days straight unsupervised.

You should also know that none of the Beaker Web APIs will work on https://hyper.reaktivo.com since they're obviously Beaker specific, this means in practice that most personal sites will break one way or another when attempting to load other hyper drives. 

Although I initially thought it would be a good idea to add polyfills for Beaker specific API, I realised it makes more sense to direct people to adopt the Beaker Browser, since that is the intention of the site.

File loading works okay though... when my hyperdrive-daemon doesn't go down.

**Comment by RangerMauve** [comment]
@reaktivo What are you using for the gateway? Is this a hyperdrive-daemon feature?

**Comment by reaktivo** [comment]
Hi @RangerMauve at the moment the site is simply a Caddy server that has been started in file serving mode on `~/Hyperdrive/Network` directly, which I guess is not optimal. I'm figuring out a way of serving without FUSE, since this is the part that keeps breaking for me.

**Reply by RangerMauve** [reply]
I've been meaning to put something together on top of dat-sdk that's similar to dat-gateway. 🤔

There's a hyperdrive-http module that's been used to view archive contents in the past. Could be time to update it with the latest file resolution code pfrazee is using for Beaker.

**Reply by reaktivo** [reply]
I hadn't seen `hyperdrive-http`, I imagine it currently doesn't work with current hyperdrives? Would love to lend of hand if needed.
